contracts/gateway/AxelarAmplifierGateway.sol

```diff
@@ -63,93 +78,117 @@ contract AxelarAmplifierGateway is IAxelarAmplifierGateway {
-
-        uint256 chainId;
-        bytes32[] memory commandIds;
-        string[] memory commands;
-        bytes[] memory params;
-
-        (chainId, commandIds, commands, params) = abi.decode(data, (uint256, bytes32[], string[], bytes[]));
+        bool allowOperatorshipTransfer = authModule.validateProof(batchHash, signedBatch.proof);
 
-        if (chainId != block.chainid) revert InvalidChainId();
+        if (chainNameHash != signedBatch.batch.chainNameHash) revert InvalidChainName();
+        if (routerHash != signedBatch.batch.routerHash) revert InvalidRouter();
 
-        uint256 commandsLength = commandIds.length;
+        Command[] memory commands = batch.commands;
 
-        if (commandsLength != commands.length || commandsLength != params.length) revert InvalidCommands();
-
-        for (uint256 i; i < commandsLength; ++i) {
-            bytes32 commandId = commandIds[i];
+        for (uint256 i; i < commands.length; ++i) {
+            Command memory command = commands[i];
+            bytes32 commandId = keccak256(bytes(command.messageId));
 
             // Ignore if commandId is already executed
             if (isCommandExecuted(commandId)) {
                 continue;
             }
 
-            bytes32 commandHash = keccak256(abi.encodePacked(commands[i]));
-
-            if (commandHash == SELECTOR_APPROVE_CONTRACT_CALL) {
-                _approveContractCall(params[i], commandId);
-            } else if (commandHash == SELECTOR_TRANSFER_OPERATORSHIP) {
+            if (command.commandType == CommandType.ApproveContractCall) {
+                _approveContractCall(commandId, command);
+            } else if (command.commandType == CommandType.TransferOperatorship) {
                 if (!allowOperatorshipTransfer) {
                     continue;
                 }
 
                 allowOperatorshipTransfer = false;
 
-                _transferOperatorship(params[i]);
+                _transferOperatorship(command.params);
             } else {
-                revert InvalidCommand(commandHash);
+                revert InvalidCommand();
             }
 
-            _axelarAmplifierGatewayStorage().commands[commandId] = true;
+            _storage().commands[commandId] = true;
 
             // slither-disable-next-line reentrancy-events
-            emit Executed(commandId);
+            emit Executed(commandId, command.messageId);
         }
     }
 
     /**********************\
     |* Internal Functions *|
     \**********************/
 
-    function _approveContractCall(bytes memory params, bytes32 commandId) internal {
-        (
-            string memory sourceChain,
-            string memory sourceAddress,
-            address contractAddress,
-            bytes32 payloadHash,
-            bytes32 sourceTxHash,
-            uint256 sourceEventIndex
-        ) = abi.decode(params, (string, string, address, bytes32, bytes32, uint256));
+    function _isContractCallApproved(
```

Comment:
Do you think we need to create internal functions if they are used only in 1 place? Keeping it inline makes contract smaller overall. And doesn't require to scroll up and down when reading the code.

Reply:
it's used twice